src/Compilers/Core/Portable/Operations/IOperation.cs

```diff
@@ -44,10 +45,16 @@ public interface IOperation
         Optional<object?> ConstantValue { get; }
 
         /// <summary>
-        /// An array of child operations for this operation.
+        /// An array of child operations for this operation. Deprecated: please use <see cref="ChildOperations"/>.
         /// </summary>
+        [Obsolete($"This API has performance penalties, please use {nameof(ChildOperations)} instead.", error: false)]
```

Comment:
>[Obsolete($"This API has performance penalties, please use {nameof(ChildOperations)} instead.", error: false)] [](http://example.com/codeflow?start=8&length=110)

The implementation simply delegates to `ChildOperations` so is boxing the only perf issue? (Depending on the callsite, boxing might be required.) 

If so, consider removing the `[Obsolete]` attribute or at least remove the warning.

Reply:
This warning is generally waht we use for perf obsoletion.  We really do not want people using teh old system and incurring double allocations when this is availble and should be suitable for all hte same purposes.

Reply:
It's not just allocations here, it's also additional virtual calls in every foreach. Obsoleteing this is intentional.